code/drug_synergy/02-drug_synergy_score_calc.R

```diff
@@ -0,0 +1,158 @@
+# Author: Run Jin
+#
+# Obtain Drug Target of Interest for each qSig output (comparing to GTEx, PBTA all or PBTA HGG)
+
+suppressPackageStartupMessages(library("optparse"))
+suppressPackageStartupMessages(library("tidyverse"))
+suppressPackageStartupMessages(library("readr"))
+suppressPackageStartupMessages(library("igraph"))
+
+#### Define Directories ----------------------------------------------------------
+root_dir <- rprojroot::find_root(rprojroot::has_dir(".git"))
+
+results_dir <- file.path(root_dir, "code", "drug_synergy", "results", "synergy_score")
+if(!dir.exists(results_dir)){
+  dir.create(results_dir, recursive = TRUE)
+}
+
+#### Parse command line options ------------------------------------------------
+
+option_list <- list(
+  make_option(c("-g","--gtex_mapped"),type="character",
+              help="Output file for drug-mapped gtex qSig subsetted subnetworks (.tsv) "),
+  make_option(c("-p","--pbta_mapped"),type="character",
+              help="Output file for drug-mapped pbta qSig subsetted subnetworks (.tsv) "),
+  make_option(c("-h","--pbta_hgg_mapped"),type="character",
+              help="Output file for drug-mapped pbta hgg subsetted subnetworks (.tsv) "),
+  make_option(c("-f","--subnetwork"),type="character",
+               help="File for subnetworks of module of interest (.tsv) "),
+  make_option(c("-b","--output_path"),type="character",
+              help="Path for results output ")
+)
+opt <- parse_args(OptionParser(option_list=option_list,add_help_option = FALSE))
+output_path <- opt$output_path 
+
+#### Read in files necessary for analyses -----------------------------------
+gtex_qSig_subnet_mapped <- readr::read_tsv(opt$gtex_mapped) %>% mutate(comparison = "gtex_qSig")
+pbta_qSig_subnet_mapped <- readr::read_tsv(opt$pbta_mapped) %>% mutate(comparison = "pbta_qSig")
+pbta_hgg_qSig_subnet_mapped <- readr::read_tsv(opt$pbta_hgg_mapped) %>% mutate(comparison = "pbta_hgg_qSig")
+
+subnetwork <- readr::read_tsv(opt$subnetwork)
+
+#### Define function that calculates synergy score -----------------------------------
+# following functions adapted from SynGeNet Publication [@doi:10.1038/s41540-019-0085-4]
+# Supplementary information downloadable as subUserFunctionsVa2.R
+
+getSynScore2 <- function(tar1, tar2, sigNet1){
```

Comment:
Agreed. 

Right now I have set up the code such that all patient level scripts go here: https://github.com/d3b-center/OMPARE/tree/master/code/patient_level_analyses and all corresponding functions go here: https://github.com/d3b-center/OMPARE/tree/master/code/patient_level_analyses/utils but they are mostly individual R scripts. Because you have been adding modules with multiple R scripts, we can keep it the way you have them structured but just moving them under `code/module_name/utils` (in this case `code/drug_synergy/utils`) and sourcing in the R scripts would be ideal.

Reply:
@komalsrathi, thanks! I have now moved the scripts to utils ;)

Reply:
@runjin326 so what you had was sort of correct, you just needed to move all the "functions" in the `code/module-name/utils/` folder and source them in your `code/module-name/01-script.R`, `code/module-name/02-script.R`.... Just like how you would for OT or OpenPBTA modules.

Reply:
An example is [enrichment script](https://github.com/d3b-center/OMPARE/blob/master/code/patient_level_analyses/gsea_enrichment.R) which takes input parameters that can be tweaked, [sources the required function ](https://github.com/d3b-center/OMPARE/blob/master/code/patient_level_analyses/gsea_enrichment.R#L24) and [calls](https://github.com/d3b-center/OMPARE/blob/master/code/patient_level_analyses/gsea_enrichment.R#L130-L139) it.

Reply:
Gotcha! Is this better? Basically I just moved the function and kept the scripts as they were.